### ISSUE TITLE
Pod fails to start, when trying to access SecurityGroup in different VPC

### ISSUE THREAD

**Issue by Voigtus** [aws/amazon-vpc-cni-k8s#1361]
**What happened**:

I want to access a RDS Database in a different AWS-Account. Therefore I used the following SecurityGroupPolicy:
```yaml
apiVersion: vpcresources.k8s.aws/v1beta1
kind: SecurityGroupPolicy
metadata:
  name: sa-sgp
  namespace: test
spec:
  serviceAccountSelector:
    matchLabels:
      role: REMOTE-accessor
  securityGroups:
    groupIds:
      - <cluster-security-group>
      - <remote-account-id>/<remote-security-group-id>
```

The result is my pod is not capable to start anymore and fails with the message:
`Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "bcd4af8f09595a54b788a1d2002066d274088f5204abbdcd634a673137cebb05" network for pod "spring-boot-demo-app-8486955796-rm6gc": networkPlugin cni failed to set up pod "spring-boot-demo-app-8486955796-rm6gc_test" network: add cmd: failed to assign an IP address to container`

I guess this happens due to this error message:
`failed to allocate branch ENI to pod: eni wrapper: unable to create network interface: InvalidParameterValue: user <my-account-id> does not own a resource <remote-account-id>/<remote-security-group-id> status code: 400, request id: 345bd2f2-f433-4957-8c92-a6b4d9403878`

Now my question: Is there another way to reference a security group in a remote VPC? Both VPC are in the same region and are peered. I followed [these instructions ](https://docs.aws.amazon.com/vpc/latest/peering/vpc-peering-security-groups.html) and I'm still not able to access the security group in the remote account.

**Environment**:
- Kubernetes version: v1.18.9-eks-d1db3c
- CNI Version: v1.7.5-eksbuild.1
- OS Image: Amazon Linux 2
- Kernel: 4.14.209-160.335.amzn2.x86_64

**Comment by jayanthvn** [comment]
Hi, 

This looks like we need add cross account permissions for remote security group. We will try this out locally and see if can provide a workaround or if it is even feasible to be supported.

Thank you!

**Comment by Voigtus** [comment]
Ok, so this feature is not part of the CNI plugin up to now? 
Good to know, because we set our hopes on exactly this use case.

Please let me know if I can be of any help.

If you have the time (and nerve) for a sanity check: but my workaround would be to allow traffic from the security group the cluster is in, to the security group where the RDS is situated in.
I know not the best idea, but beggars can't be choosers.

**Comment by SaranBalaji90** [comment]
@Voigtus i have suggestion let me know if this would work for your case. Created a security group in the VPC where you have your worker node and provide that in the CRD. As a second step, add Ingress rule on your RDS security group to allow access from this newly created security group.

**Comment by Voigtus** [comment]
@SaranBalaji90 Sounds like a straightforward plan. I will evaluate this step and get back to you the scond I get a result.

Thank you for your assistance. 😊

**Comment by jayanthvn** [comment]
Hi @Voigtus 

Can you please let us know if you got a chance to verify the suggestion by @SaranBalaji90

**Comment by SaranBalaji90** [comment]
Closing this as workaround might unblock, if not myself or @jayanthvn will reopen this issue. Thanks.

**Comment by Voigtus** [comment]
I'm very sorry I wasn't capable to answer on time.
Yes @SaranBalaji90 that workaround worked like a charm, but created an "uprising" wether we should reduce complexicity by merging these two accounts in one. 

Thank you for your patience and this workaround. It saved us a lot of time and hustle. And created new ones, just as it should be 😅
